hq/app/service/CacheService.scala

```diff
@@ -0,0 +1,58 @@
+package service
+
+import akka.actor.ActorSystem
+import com.gu.Box
+import config.Config
+import model._
+import play.api.Configuration
+import rx.lang.scala.Observable
+import utils.attempt.{Attempt, FailedAttempt}
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+
+class CacheService(config: Configuration) {
+  private val accounts = Config.getAwsAccounts(config)
+
+  def update[T](actorSystem: ActorSystem)(awsWorkerFn: (List[AwsAccount]) => Attempt[T])(boxUpdaterFn: Attempt[T] => Attempt[Unit])(implicit ec: ExecutionContext) = {
+    val cacheInterval = Config.getAwsApiCacheInterval(config)
+    val subscription = Observable.interval(10.millisecond, cacheInterval).doOnEach { _ =>
+      boxUpdaterFn(awsWorkerFn(accounts))
+    }.subscribe
+
+    actorSystem.registerOnTermination(
+      subscription.unsubscribe()
+    )
+  }
+}
+
+object CacheService {
+  type EitherFlaggedSgs = Either[FailedAttempt, List[(SGOpenPortsDetail, Set[SGInUse])]]
+  type FlaggedSgs = List[(AwsAccount, EitherFlaggedSgs)]
+  type ExposedKeys = List[Either[FailedAttempt, (AwsAccount, TrustedAdvisorDetailsResult[ExposedIAMKeyDetail])]]
+  type CredentialReport = Seq[(AwsAccount, Either[FailedAttempt, CredentialReportDisplay])]
+
+  private val exposedKeysBox = Box[ExposedKeys](List.empty)
+  private val credentialReportBox = Box[CredentialReport](List.empty)
+  private val sgsBox = Box[FlaggedSgs](List.empty)
```

Comment:
It might be worth revisiting these boxes (another PR?) once we have a clear idea for our data structures. Ideally, the boxes would store the data in the most helpful shape ready for use. For example:

The ideal data structure of the sgsBox might be:
```
Box[Map[String, Either[FailedAttempt, List[(SGOpenPortsDetail, Set[SGInUse])]]]]]
```
Would this allow us to skip the filtering step in the controller (for single accounts) and minimise the amount transformation required?